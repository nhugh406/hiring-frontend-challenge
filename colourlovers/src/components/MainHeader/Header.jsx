import React, { forwardRef, useImperativeHandle, useState } from 'react';
import './Styles.scss';
const Header = forwardRef((props, ref) => {
  const time = () => {
    const d = new Date();
    const h = d.getHours();
    const m = d.getMinutes();
    const form = `${h > 12 ? h - 12 : h}:${m} ${h > 12 ? 'pm' : 'am'}`;
    setUpdateTime(form);
  };

  const [updateTime, setUpdateTime] = useState();

  useImperativeHandle(ref, () => ({
    updateData() {
      time();
    },
  }));

  return (
    <div className="main-header">
      <h1>
        Colour Lovers. <span className="bold-text">Live.</span>
      </h1>
      <h3 className="update-timer">Last Updated at {updateTime}</h3>
    </div>
  );
});

export default Header;
