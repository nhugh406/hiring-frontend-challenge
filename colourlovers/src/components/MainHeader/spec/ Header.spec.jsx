import React from 'react';
import { render } from '@testing-library/react';

import Header from '../Header';

describe('Colour Family Component Tests', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});
