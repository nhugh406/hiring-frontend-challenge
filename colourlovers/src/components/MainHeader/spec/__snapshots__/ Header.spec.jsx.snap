// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colour Family Component Tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="main-header"
  >
    <h1>
      Colour Lovers. 
      <span
        class="bold-text"
      >
        Live.
      </span>
    </h1>
    <h3
      class="update-timer"
    >
      Last Updated at 
    </h3>
  </div>
</DocumentFragment>
`;
